### PR TITLE
chore(RHTAPREL-645): add release overlays

### PR DESCRIPTION
**Pull request by scoheb**
- new development overlay (default for infra-deployment)
- new staging overlay
- new production overlay
- updated preview.sh for new kustomization.yaml location

Related PR: https://github.com/redhat-appstudio/tenants-config/pull/82

### CODE REVIEW

File: hack/preview.sh
Comment (scoheb) [reply hack/preview.sh]: yep. I am learning!

fixed!

File: components/release/staging/kustomization.yaml
Comment (scoheb) [reply components/release/staging/kustomization.yaml:11]: yessir! fixed!

File: components/release/base/kustomization.yaml
Comment (johnbieren) [inline components/release/base/kustomization.yaml:3]: Why is there a directory `release-pipeline-role` containing a single yaml in it and a `kustomization` file instead of just pointing to the yaml file here? Same question for `authentication`

Comment (scoheb) [reply components/release/base/kustomization.yaml:3]: do you mean to say they should be in `1 folder` with `1 kustomization.yaml` file?

> I am following what the `has` team has been using up to now. See https://github.com/redhat-appstudio/infra-deployments/tree/main/components/has/base

or do you mean to say why are we even using `kustomization.yaml` files anyways?

> I had issues including them in overlays without a `kustomization.yaml` file

Comment (johnbieren) [reply components/release/base/kustomization.yaml:3]: I am saying why not replace `release-pipeline-role` with `release-pipeline-resources-clusterrole.yaml` and move the yaml up a dir and not have the extra `release-pipeline-role` dir

Comment (scoheb) [reply components/release/base/kustomization.yaml:3]: > I am saying why not replace `release-pipeline-role` with `release-pipeline-resources-clusterrole.yaml` and move the yaml up a dir and not have the extra `release-pipeline-role` dir

I can do that!

Comment (scoheb) [reply components/release/base/kustomization.yaml]: fixed.

File: argo-cd-apps/overlays/development/kustomization.yaml
Comment (johnbieren) [inline argo-cd-apps/overlays/development/kustomization.yaml:1]: Don't we want a change like this for staging?

File: components/release/base/kustomization.yaml
Comment (johnbieren) [reply components/release/base/kustomization.yaml]: awesome, thanks

File: argo-cd-apps/overlays/development/kustomization.yaml
Comment (scoheb) [reply argo-cd-apps/overlays/development/kustomization.yaml:1]: staging does not have such a patch. See `argo-cd-apps/overlays/staging/kustomization.yaml`

File: components/release/production/kustomization.yaml
Comment (scoheb) [reply components/release/production/kustomization.yaml:7]: > So we are saying this is not needed in the development setup?

I do not think we need to up the resource for the manager in the dev setup.

> I am also a little confused about the switch from patches: -path to components: - I guess it works about the same?

I believe so. not 100% sure myself. I am copying what has is doing.

Comment (johnbieren) [reply components/release/production/kustomization.yaml:7]: ack

File: argo-cd-apps/overlays/development/kustomization.yaml
Comment (johnbieren) [reply argo-cd-apps/overlays/development/kustomization.yaml:1]: ah my bad

**Review by johnbieren — APPROVED** [review]
Can't say I understand every line of the change perfectly, but I do most of it and it lgtm

**Comment by scoheb** [comment]
> Can't say I understand every line of the change perfectly, but I do most of it and it lgtm

Thanks. @davidmogar has more kustomize experience than me...so I'll wait for his approval.

File: argo-cd-apps/overlays/production/kustomization.yaml
Comment (johnbieren) [inline argo-cd-apps/overlays/production/kustomization.yaml:123]: rebase error

**Review by johnbieren — CHANGES_REQUESTED** [review]

**Pull request merged**
